### PR TITLE
Changed displaying modifiers

### PR DESCRIPTION
**Pull request by przepompownia**
- show more modifiers if available for method,
for example "!_ @" for final public static;
- configuration between modifier and its sign was separated;

### CODE REVIEW

**Comment by przepompownia** [comment]
What do you think about this change made by the way another debugging?

File: php/PHPCD.php [hunk 1]
```diff
@@ -319,34 +334,42 @@ private function getMethodInfo($method, $pattern = null)
...
-        return "$modifier $static";
+    private function getModifiers($reflection)
+    {
+        $signs = '';
+
+        $modifiers = $reflection->getModifiers();
+        $symbols = $this->getModifierSymbols();
+
+        foreach ($symbols as $number => $sign) {
+            if ($number & $modifiers) {
```
Comment (lvht) [inline php/PHPCD.php]: :+1:

File: php/PHPCD.php [hunk 2]
```diff
@@ -2,7 +2,19 @@
 
 class PHPCD extends RpcServer
 {
-    public function info($class_name, $pattern, $mode) {
+    /**
+     *  @param array Map between modifier numbers and displayed symbols
+     */
+    private $modifierSymbols = [
```
Comment (lvht) [inline php/PHPCD.php]: the order will make sense. In my opinion, `$modifierSymbols` should be
```
        ReflectionMethod::IS_FINAL      => '!',
        ReflectionMethod::IS_STATIC     => ' @'
        ReflectionMethod::IS_PRIVATE    => '-',
        ReflectionMethod::IS_PROTECTED  => '#',
        ReflectionMethod::IS_PUBLIC     => '+',
```

File: php/PHPCD.php [hunk 3]
```diff
@@ -319,34 +334,42 @@ private function getMethodInfo($method, $pattern = null)
             return $param->getName();
         }, $method->getParameters());
 
-        $modifier = $this->getModifier($method);
+        $modifier = $this->getModifiers($method);
 
         return [
             'word' => $name,
-            'abbr' => "$modifier $name (" . join(', ', $params) . ')',
+            'abbr' => sprintf("%s\t\t%s (%s)", $modifier, $name, join(', ', $params)),
```
Comment (lvht) [inline php/PHPCD.php]: `%s\t\t%s (%s)" will be not suitable for some case like
```
final static private function hehe() {}
```

So, what about to use `%4s %s (%s)`?

**Comment by lvht** [comment]
Using the `getModifiers` method and bitwise operation is fantastic :+1:

File: php/PHPCD.php (hunk 3 above)
Comment (przepompownia) [inline php/PHPCD.php]: Max 3 characters should be enough. Use of padding - the most :)

File: php/PHPCD.php (hunk 2 above)
Comment (przepompownia) [inline php/PHPCD.php]: On the one hand, I kept the present order ("$modifier $static"), on the other - I follow PSR-2 suggestion concerning order (https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#1-overview).

**Comment by imaben** [comment]
@przepompownia :+1:

**Pull request merged**